test/transferFrom.js

```diff
@@ -3,13 +3,35 @@ const InkProtocol = artifacts.require("./mocks/InkProtocolMock.sol")
 
 contract("InkProtocol", (accounts) => {
   let protocol
+  let sender
+  let recipient
+  let agent
+  let amount
 
   beforeEach(async () => {
     protocol = await InkProtocol.new()
+    sender = accounts[1]
+    recipient = accounts[2]
+    agent = accounts[3]
+    amount = 10
   })
 
   describe("#transferFrom()", () => {
-    it("fails when recipient is the protocol")
-    it("succeeds when recipient is another address")
+    it("fails when recipient is the protocol", async () => {
+      recipient = protocol.address
+
+      await $util.assertVMExceptionAsync(protocol.transferFrom(sender, recipient, amount))
+    })
+
+    it("succeeds when recipient is another address", async () => {
+      await protocol.transfer(sender, amount)
+      let originalSenderBalance = await $util.getBalance(sender, protocol)
+
+      await protocol.approve(agent, amount, { from: sender })
+      await protocol.transferFrom(sender, recipient, amount, { from: agent })
+
+      assert.equal(await $util.getBalance(sender, protocol), originalSenderBalance - amount)
```

Comment:
No need to lookup `originalSenderBalance` in line 28. You know it should be at the end: 0.